client/src/components/inventory/HardwareList.tsx

```diff
@@ -54,97 +57,138 @@ const sampleData = [
-                              id={item.id}
-                              addItem={this.props.handleAddItem} // prop that invokes the handleAddItem method of parent container to update its state
-                              qtyUpdate={this.props.qtyUpdate} // this prop is the object whose request has been cancelled
-                              user={this.props.user}
-                />))}
-        </Item.Group>);
-        const loading = (<Item.Group>
-            <PlaceholderItem/>
-            <PlaceholderItem/>
-            <PlaceholderItem/>
-        </Item.Group>);
-
-        return (
-                <div>
+    public render() {
+        return <Query
+            query={gql`
+                query {
+                    items {
+                        id
+                        item_name
+                        description
+                        imageUrl
+                        category
+                        totalAvailable
+                        maxRequestQty
+                        hidden
+                        approvalRequired
+                        returnRequired
+                        owner
+                    }
+                }
+            `}>
+            {({loading, error, data}: any) => {
+                if (error) {
+                    console.log("error", error);
+                }
+                // TODO: come back to this
+                if (loading) {
+                    return (
+                        <div>
+                            <Header>Inventory</Header>
+                            const loadingGroup = (<Item.Group>
+                            <PlaceholderItem/>
+                            <PlaceholderItem/>
+                            <PlaceholderItem/>
+                        </Item.Group>);
+                        </div>
+                    );
+                }
+                if (error) {
+                    return <Message negative>
+                        <Message.Header>Error displaying hardware inventory</Message.Header>
+                        <p>Try refreshing the page. If that doesn't work, contact a member of the HackGT Team for
+                            assistance.</p>
+                    </Message>;
+                }
+
+                let noRequestsMessageText = "";
+
+                if (!this.props.requestsEnabled) {
+                    noRequestsMessageText = "Hardware checkout requests can't be made at this time.";
+                } else if (this.props.requestsEnabled && !this.props.user) {
+                    noRequestsMessageText = "Sign in to request hardware.";
+                }
+
+                const noRequestsMessage = this.props.requestsEnabled && !this.props.user ? (<Message
+                    title="View-only inventory"
+                    warning icon>
+                    <Icon name="warning sign"/>
+                    {noRequestsMessageText}
+                </Message>) : "";
+                console.log(loading, error, data);
```

Comment:
careful with logs on master, I don't mind deeply but don't forget

Reply:
Removed it